src/pbrt/media.h

```diff
@@ -346,50 +346,47 @@ class CuboidMedium {
     Point3i gridResolution;
 };
 
-// UniformGridMediumProvider Definition
-class UniformGridMediumProvider {
+
+// GridMediumProvider Definition
+class GridMediumProvider {
   public:
-    // UniformGridMediumProvider Public Methods
-    UniformGridMediumProvider(const Bounds3f &bounds,
-                              pstd::optional<SampledGrid<Float>> density,
-                              pstd::optional<SampledGrid<Float>> sigma_a,
-                              pstd::optional<SampledGrid<Float>> sigma_s,
-                              pstd::optional<SampledGrid<RGBUnboundedSpectrum>> rgb,
-                              Spectrum Le, SampledGrid<Float> LeScale, Allocator alloc);
+    // GridMediumProvider Public Methods
+    GridMediumProvider(const Bounds3f &bounds, SampledGrid<Float> density,
+                       pstd::optional<SampledGrid<Float>> temperature, Spectrum Le,
+                       SampledGrid<Float> LeScale, Allocator alloc);
 
-    static UniformGridMediumProvider *Create(const ParameterDictionary &parameters,
-                                             const FileLoc *loc, Allocator alloc);
+    static GridMediumProvider *Create(const ParameterDictionary &parameters,
+                                      const FileLoc *loc, Allocator alloc);
 
     std::string ToString() const;
 
     PBRT_CPU_GPU
     const Bounds3f &Bounds() const { return bounds; }
 
     PBRT_CPU_GPU
-    bool IsEmissive() const { return Le_spec.MaxValue() > 0; }
+    bool IsEmissive() const {
+        if (temperatureGrid)
+            // TODO Assume the grid's max value is greater than 100.f?
+            return true;
+        return Le_spec.MaxValue() > 0;
+    }
 
     PBRT_CPU_GPU
     SampledSpectrum Le(Point3f p, const SampledWavelengths &lambda) const {
         Point3f pp = Point3f(bounds.Offset(p));
+        if (temperatureGrid) {
+            Float temp = temperatureGrid->Lookup(pp);
+            if (temp <= 100.f)
```

Comment:
Out of curiosity, why this check?

Reply:
Good question, I was attempting to match what  NanoVDBMediumProvider::Le() did.
https://github.com/mmp/pbrt-v4/blob/77d367204ad7c73eb5072c62f5e6f12604f0574d/src/pbrt/media.h#L625-L626

Reply:
Got it. I'd suggest that should be in the same category of "up to the user if they want that or not, since they're generating the input," but in any case let's get this landed and I'll revisit that in a bit...